src/components/smime/SMimeCertificateModal.vue

```diff
@@ -0,0 +1,233 @@
+						</tr>
+					</thead>
+					<tbody>
+						<tr v-for="certificate in parsedCertificates" :key="certificate.id">
+							<td>{{ certificate.commonName }}</td>
+							<td>{{ certificate.emailAddress }}</td>
+							<td>{{ moment(certificate.notAfter).format('LL') }}</td>
+							<td>
+								<NcButton type="secondary" @click="deleteCertificate(certificate.id)">
+									<template #icon>
+										<DeleteIcon :size="20" />
+									</template>
+									{{ t('mail', 'Delete') }}
+								</NcButton>
+							</td>
+						</tr>
+					</tbody>
+				</table>
+
+				<div class="certificate-modal__list__actions">
+					<NcButton type="primary" @click="showImportScreen = true">
+						{{ t('mail', 'Import certificate') }}
+					</NcButton>
+				</div>
+			</div>
+			<form
+				v-else
+				class="certificate-modal__import"
+				@submit.prevent="uploadCertificate">
+				<h2>{{ t('mail', 'Import S/MIME certificate') }}</h2>
+
+				<fieldset>
+					<label for="certificate">{{ t('mail', 'Certificate') }}</label>
+					<input
+						id="certificate"
+						ref="certificate"
+						type="file"
+						@change="certificate = $event.target.files[0]">
+				</fieldset>
+
+				<fieldset>
+					<label for="private-key">{{ t('mail', 'Private key') }}</label>
+					<input
+						id="private-key"
+						ref="privateKey"
+						type="file"
+						@change="privateKey = $event.target.files[0]">
+				</fieldset>
+
+				<div class="certificate-modal__import__actions">
+					<NcButton
+						type="tertiary-no-background"
+						@click="resetImportForm">
+						{{ t('mail', 'Back') }}
+					</NcButton>
+					<NcButton
+						type="primary"
+						native-type="submit"
+						:disabled="loading || !inputFormIsValid">
+						{{ t('mail', 'Submit') }}
+					</NcButton>
+				</div>
+			</form>
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { NcButton, NcModal } from '@nextcloud/vue'
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import logger from '../../logger'
+import forge from 'node-forge'
+import moment from '@nextcloud/moment'
+import DeleteIcon from 'vue-material-design-icons/Delete'
+
+export default {
+	name: 'SMimeCertificateModal',
+	components: {
+		NcModal,
+		NcButton,
+		DeleteIcon,
+	},
+	data() {
+		return {
+			moment,
+			showImportScreen: false,
+			loading: false,
+			certificate: undefined,
+			privateKey: undefined,
+		}
+	},
+	computed: {
+		...mapGetters({
+			certificates: 'getSMimeCertificates',
+		}),
+		inputFormIsValid() {
+			return !!this.certificate
+		},
+		parsedCertificates() {
+			logger.debug('certs', { certs: this.certificates })
+			const certs = this.certificates.map(certificate => {
+				const parsed = forge.pki.certificateFromPem(certificate.certificate)
+				return {
+					...certificate,
+					notAfter: parsed.validity.notAfter,
+					commonName: parsed.subject.getField('CN').value,
+					parsed,
+				}
+			})
+			logger.debug('parsedCertificates', { certs })
```

Comment:
let's redact any sensitive values here because the logs might be written to disk or sent to a logging service

Reply:
This line can go. It's just a debugging leftover.